feature(resolver): Add support for get parameters

Cawllec opened the pull request:
- Adds support for `get` parameters in the basic resolver
- Retains support for `$_POST` and other method parameters.

fractalwrench commented on tests/RequestTest.php:
```diff
@@ -103,6 +103,23 @@ public function testGetMetaDataWithPost()
 
+    public function testGetMetaDataWithGet()
+    {
+        $_SERVER['REQUEST_METHOD'] = 'GET';
```
What would happen with lesser used request methods like `HEAD` or `PATCH`? Is it worth covering them?

Cawllec replied:
They don't have server variables and the data should be dropped into either the `json_decode` section due to `application/json` being present, or the `parse_str` section if not.  I'll do some manual testing on this to verify these cases are covered however.

GrahamCampbell commented on src/Request/BasicResolver.php:
```diff
@@ -71,19 +71,22 @@ protected static function getRequestHeaders(array $server)
 
-        $result = $post ?: static::parseInput($server, static::readInput());
+        if (isset($server['REQUEST_METHOD']) && strtoupper($server['REQUEST_METHOD']) === 'GET') {
+            $result = $_GET;
```
Access to superglobals was intentionally omitted from anything other than the resolve method. Perhaps put it back there, so that this method is not messing with global state?

Cawllec replied:
Does it matter if the method is only being called statically from within the `resolve` method?

kattrali commented on src/Request/BasicResolver.php:
```diff
@@ -16,7 +16,7 @@ public function resolve()
                 empty($_COOKIE) ? [] : $_COOKIE,
                 static::getRequestHeaders($_SERVER),
-                static::getInputParams($_SERVER, $_POST));
+                static::getInputParams($_SERVER));
```
Checking the request method should instead happen here, and pass the correct variable array (get vs post variables) into `getInputParams`. It reduces the amount of "magic" going on throughout.

Cawllec replied:
In the current implementation if `$_GET` is empty in a `GET` request the method will attempt to pull body parameters from `php://input`, something that it shouldn't do for `GET` requests. 
I'd suggest splitting `getInputParams` into multiple methods to handle `GET` and other request methods separately, or add an argument that determines whether the `php://input` source should be read or not.

kattrali replied:
Correct, checking whether to read from `php://input` should be done by request type, either in a single place or via a flag. I'm not suggesting that input should be read for GET requests.

Cawllec commented:
Ok, I've refactored it somewhat so that the superglobals `$_SERVER`, `$_POST`, and `$_GET` are accessed only in the `resolve` method, and passed through to `getInputParams` as before.

I've added a flag, `$is_get` to the method in order to determine whether the `parseInput` methods should be called.

The result doesn't effect the output, so the tests have remained the same.
